src/services/services.ts

```diff
@@ -2788,18 +2788,37 @@ namespace ts {
         return node && node.parent && node.parent.kind === SyntaxKind.PropertyAccessExpression && (<PropertyAccessExpression>node.parent).name === node;
     }
 
+    function climbPastPropertyAccess(node: Node) {
+        return isRightSideOfPropertyAccess(node) ? node.parent : node;
+    }
+
     function isCallExpressionTarget(node: Node): boolean {
-        if (isRightSideOfPropertyAccess(node)) {
-            node = node.parent;
-        }
-        return node && node.parent && node.parent.kind === SyntaxKind.CallExpression && (<CallExpression>node.parent).expression === node;
+        return !!getCallOrNewExpressionWorker(node, SyntaxKind.CallExpression);
     }
 
     function isNewExpressionTarget(node: Node): boolean {
-        if (isRightSideOfPropertyAccess(node)) {
-            node = node.parent;
+        return !!getCallOrNewExpressionWorker(node, SyntaxKind.NewExpression);
+    }
+
+    function getCallOrNewExpressionTargetingNode(node: Node): CallExpression | NewExpression | undefined {
+        return <CallExpression>getCallOrNewExpressionWorker(node, SyntaxKind.CallExpression) || <NewExpression>getCallOrNewExpressionWorker(node, SyntaxKind.NewExpression);
+    }
+
+    function tryGetCalledDeclaration(typeChecker: TypeChecker, node: Node): SignatureDeclaration | undefined {
+        const callOrNewExpression = getCallOrNewExpressionTargetingNode(node);
+        if (callOrNewExpression) {
+            const signature = typeChecker.getResolvedSignature(callOrNewExpression);
+            return signature.declaration;
         }
-        return node && node.parent && node.parent.kind === SyntaxKind.NewExpression && (<CallExpression>node.parent).expression === node;
+    }
+
+    function getCallOrNewExpressionWorker(node: Node, kind: SyntaxKind): Node | undefined {

```

Comment:
Change the type of `kind` to `SyntaxKind.CallExpression | SyntaxKind.NewExpression`